### PR TITLE
Fix additional mypy errors and expand CI checks to more files

### PR DESCRIPTION
**Pull request by correctmost**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description
Clean-up of some of the simpler mypy errors.

The import changes are to fix errors like this:
```
astroid/brain/brain_namedtuple_enum.py:16: error: Module "astroid" does not explicitly export attribute "inference_tip"  [attr-defined]
```

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Wow, amazing that you were able to activate so much files at once and use ``nodes.X`` type import style without triggering a cyclic import. This is great work, and very important work on the path to make astroid really faster in the future 👏. (refs #2014)

I'm going to wait for a review by Daniel, Marc or Jacob that are more knowledgeable in astroid typing than myself but this is 🔥

File: pyproject.toml
Comment (DanielNoord) [inline pyproject.toml:70]: Can we use `globs` here? And just enable for all of `brain`?

File: astroid/brain/brain_hypothesis.py
Comment (DanielNoord) [inline astroid/brain/brain_hypothesis.py:44]: Shouldn't we do asserts here? Instead of ignoring?

File: astroid/brain/brain_hypothesis.py
Comment (DanielNoord) [inline astroid/brain/brain_hypothesis.py:46]: I think it is better to actually do this assert always. `mypy` is showing us we have unsafe code, we should make it safe by asserting our assumption so we can see where they don't hold and handle it accordingly.

**Review by DanielNoord — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**
